pkg/domain/infra/abi/quadlet.go

```diff
@@ -0,0 +1,764 @@
+	logrus.Debugf("Going to install Quadlet to directory %s", installDir)
+
+	_, err = os.Stat(installDir)
+	// Make the directory if it doesn't exist
+	if err != nil && errors.Is(err, fs.ErrNotExist) {
+		if err := os.MkdirAll(installDir, 0755); err != nil {
+			return nil, fmt.Errorf("unable to create Quadlet install path %s: %w", installDir, err)
+		}
+	} else if err != nil {
+		return nil, fmt.Errorf("unable to stat Quadlet install path %s: %w", installDir, err)
+	}
+
+	installReport := entities.QuadletInstallReport{
+		InstalledQuadlets: make(map[string]string),
+		QuadletErrors:     make(map[string]error),
+	}
+
+	assetFile := ""
+	paths := pathsOrURLs
+	if len(pathsOrURLs) > 0 && !strings.HasPrefix(pathsOrURLs[0], "http://") && !strings.HasPrefix(pathsOrURLs[0], "https://") {
+		// Check if first path is dir, this is an APP
+		info, err := os.Stat(pathsOrURLs[0])
+		if err != nil {
+			return nil, fmt.Errorf("unable to stat Quadlet path %s: %w", pathsOrURLs[0], err)
+		}
+		if info.IsDir() {
+			// If it's a directory, then read all files and add it to paths
+			entries, err := os.ReadDir(pathsOrURLs[0])
+			if err != nil {
+				return nil, fmt.Errorf("unable to read Quadlet dir %s: %w", pathsOrURLs[0], err)
+			}
+			redoPaths := []string{}
+			for _, entry := range entries {
+				redoPaths = append(redoPaths, filepath.Join(pathsOrURLs[0], entry.Name()))
+			}
+			redoPaths = append(redoPaths, pathsOrURLs[1:]...)
+			paths = redoPaths
+			// treat all file in this session as part of one app.
+			assetFile = "." + filepath.Base(pathsOrURLs[0]) + ".app"
+		}
+	}
+
+	// Loop over all given URLs
+	for _, toInstall := range paths {
+		validateQuadletFile := false
+		if assetFile == "" {
+			assetFile = "." + filepath.Base(toInstall) + ".asset"
+			validateQuadletFile = true
+		}
+		switch {
+		case strings.HasPrefix(toInstall, "http://") || strings.HasPrefix(toInstall, "https://"):
+			r, err := http.Get(toInstall)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = fmt.Errorf("unable to download URL %s: %w", toInstall, err)
+				continue
+			}
+			defer r.Body.Close()
+			quadletExtension := getFileExtension(r)
+			// It's a URL. Pull to temporary file.
+			tmpFile, err := os.CreateTemp("", "quadlet-dl-*"+quadletExtension)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = fmt.Errorf("unable to create temporary file to download URL %s: %w", toInstall, err)
+				continue
+			}
+			defer func() {
+				tmpFile.Close()
+				if err := os.Remove(tmpFile.Name()); err != nil {
+					logrus.Errorf("Unable to remove temporary file %q: %v", tmpFile.Name(), err)
+				}
+			}()
+			_, err = io.Copy(tmpFile, r.Body)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = fmt.Errorf("populating temporary file: %w", err)
+				continue
+			}
+			installedPath, err := ic.installQuadlet(ctx, tmpFile.Name(), "", installDir, assetFile, validateQuadletFile)
```

Comment:
Again this makes no sense,
```
podman quadlet install https://github.com/containers/podman/blob/main/test/e2e/quadlet/basic.container
/home/pholzing/.config/containers/systemd/quadlet-dl-3360325579.container
```

This results in a quadlet with a completely random name which is just nonsense, it should take the filename of the URL. That is after all why installQuadlet() has the third argument to specify the name we want.